apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.ts

```diff
@@ -126,13 +128,50 @@ router.post(
       ],
       z.object({
         user_id: IdSchema,
-        has_access: z.boolean(),
+        has_legacy_access: z.boolean(),
+        is_modern_publishing: z.boolean(),
+        publishing_start_date: DateFromISOString.nullable(),
+        publishing_end_date: DateFromISOString.nullable(),
+        latest_publishing_extension_date: DateFromISOString.nullable(),
```

Comment:
👀 I have some thoughts about all this:

- The name of the sproc is inaccurate now. Maybe `users_select_or_insert_lti` is more appropriate?
- Is there any reason we can't leverage e.g. `constructCourseOrInstanceContext` and `ensureEnrollment` here?

I say this with full knowledge that we had previously said that we'd just let this rot, but I think you've actually given us some really nice primitives that we can pretty easily compose to do the right thing the right way here.

Reply:
Renamed in e528ce9172eea5addf0207f3aa1829ddcf3c460b.